gui/src/app/Scripting/Analysis/useAnalysisState.ts

```diff
@@ -0,0 +1,75 @@
+import {
+  InterpreterStatus,
+  isInterpreterBusy,
+} from "@SpScripting/InterpreterTypes";
+import { StanRun } from "@SpStanSampler/useStanSampler";
+import { useEffect, useMemo, useRef, useState } from "react";
+
+export type GlobalDataForAnalysis = {
+  draws: number[][];
+  paramNames: string[];
+  numChains: number;
+};
+
+// A custom hook to share logic between the Python and R analysis windows
+// This contains the output div refs, the interpreter state, and the data from
+// the latest run.
+const useAnalysisState = (latestRun: StanRun) => {
+  const consoleRef = useRef<HTMLDivElement | null>(null);
+  const imagesRef = useRef<HTMLDivElement | null>(null);
+
+  useEffect(() => {
+    if (imagesRef.current) {
+      imagesRef.current.innerHTML = "";
+    }
+    if (consoleRef.current) {
+      consoleRef.current.innerHTML = "";
+    }
```

Comment:
This shows up here and in `AnalysisPyWindow.tsx`, maybe pop it in a function?

Reply:
Done